### PR TITLE
Add httpclient.`$`(MultipartData)

### LINKED ISSUE #11863: MultipartData needs $
### Summary
- `httpclient.MultipartData` needs a `$` to stringuify. 
- Create a ```httpclient.`$`*(data: httpclient.MultipartData): string```.
- `MultipartData` is just a `seq[string]` so it should not be too complex to add.

Actual error of doing `echo` on a `MultipartData`:
```nim
Error: type mismatch: got <MultipartData>
expression: `$`(multipart_data)
```

Its useful for debugging and learning.
:slightly_smiling_face:

### PR DESCRIPTION
**Pull request by konradmb**
Fixes  #11863

### CODE REVIEW

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -425,6 +425,41 @@ proc newMultipartData*: MultipartData =
   ## Constructs a new ``MultipartData`` object.
   MultipartData(content: @[])
 
+proc parseMultipartDataString(item: string): string =
+  var itemPart: seq[string] = item.
```
Comment (juancarlospaco) [inline lib/pure/httpclient.nim]: Splitting the line on an ending dot (period) like this I think is not used on the Style of Nim.
Maybe run `nimpretty` on it and then adjust it manually :grey_question: 
:thinking:

Comment (konradmb) [reply lib/pure/httpclient.nim]: nimpretty breaks it after `(`

**Comment by juancarlospaco** [comment]
Both proc need documentation comments.

Would be cool to move `parseMultipartDataString()` to [parseutils](https://nim-lang.github.io/Nim/parseutils.html) and make it public.
:thinking:

**Comment by konradmb** [comment]
> Both proc need documentation comments.
Ok, but `$` is self-explaining.

> Would be cool to move `parseMultipartDataString()` to [parseutils](https://nim-lang.github.io/Nim/parseutils.html) and make it public.

I disagree. I made this proc only to make code clearer. It works only in this case - when parsing string from corresponding add() function for MultipartData. I'm sure it would bug out on in-the-wild data.

**Comment by Araq** [comment]
Don't use ``split`` for parsing. Use a hand written parser or the ``strscans`` module.

**Comment by konradmb** [comment]
> Don't use `split` for parsing.

Ok, I'll rewrite it, but why exactly? Is it because split is slow?

**Comment by dom96** [comment]
Why even parse this stuff? Isn't it good enough to just print out the string you already have?

**Comment by konradmb** [comment]
For this:
```nim
var data = newMultipartData()
data["output"] = "soap12"
data["uploaded_file"] = ("test.html", "text/html",
"<html><head></head><body><p>test</p></body></html>")
```

Output:
```nim
@["Content-Disposition: form-data; name=\"output\"\c\n\c\nsoap12\c\n", "Content-Disposition: form-data; name=\"uploaded_file\"; filename=\"test.html\"\c\nContent-Type: text/html\c\n\c\n<html><head></head><body><p>test</p></body></html>\c\n"]
```

It's not exactly human-readable :smile:

**Comment by Araq** [comment]
> Ok, I'll rewrite it, but why exactly? Is it because split is slow?

No it is because ``split`` is totally incapable of getting any kind of escaping/quoting rules right.

**Comment by narimiran** [comment]
Should this be closed in favour of https://github.com/nim-lang/Nim/pull/12412 ?

**Comment by metagn** [comment]
#12707 was superseded by #12412 and got merged

**Pull request closed without merging**